### PR TITLE
adds encryption config observer

### PR DESCRIPTION
**Pull request by p0lyn0mial**

### CODE REVIEW

File: pkg/operator/configobservation/configobservercontroller/observe_config_controller.go [hunk 1]
```diff
@@ -62,13 +64,15 @@ func NewConfigObserver(
 				SchedulerLister:       configInformer.Config().V1().Schedulers().Lister(),
 
 				ConfigmapLister:              kubeInformersForNamespaces.ConfigMapLister(),
+				SecretLister_:                kubeInformersForNamespaces.InformersFor(operatorclient.TargetNamespace).Core().V1().Secrets().Lister(),
```
Comment (sttts) [inline pkg/operator/configobservation/configobservercontroller/observe_config_controller.go]: why underscore?

File: pkg/operator/configobservation/configobservercontroller/observe_config_controller.go [hunk 2]
```diff
@@ -87,6 +91,10 @@ func NewConfigObserver(
 			apiserver.ObserveUserClientCABundle,
 			apiserver.ObserveAdditionalCORSAllowedOrigins,
 			auth.ObserveAuthMetadata,
+			encryption.NewEncryptionConfigObserver(
+				operatorclient.TargetNamespace,
+				[]string{"apiServerArguments", "encryption-provider-config"},
```
Comment (sttts) [inline pkg/operator/configobservation/configobservercontroller/observe_config_controller.go]: why is this a parameter? Put it deep into the code where it is used.

Comment (sttts) [reply pkg/operator/configobservation/configobservercontroller/observe_config_controller.go]: Or is it different for openshift-apiserver? (worth checking before following my suggestion :-) )

File: pkg/operator/configobservation/interfaces.go [hunk 3]
```diff
@@ -9,7 +9,9 @@ import (
 	"github.com/openshift/library-go/pkg/operator/resourcesynccontroller"
 )
 
-var _ cloudprovider.InfrastructureLister = &Listers{}
+var (
+	_ cloudprovider.InfrastructureLister = Listers{}
+)
```
Comment (sttts) [inline pkg/operator/configobservation/interfaces.go]: why this change? I like oneliners.

File: pkg/operator/encryption/observe_encryption_config.go [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+const (
+	encryptionConfFilePath = "/etc/kubernetes/static-pod-resources/secrets/encryption-config/encryption-config"
```
Comment (sttts) [inline pkg/operator/encryption/observe_encryption_config.go]: this will be different for openshift-apiserver. It has to be a parameter to the observer constructor.

File: pkg/operator/encryption/observe_encryption_config.go [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+		observedConfig := map[string]interface{}{}
+
+		encryptionConfigSecret, err := listers.SecretLister().Secrets(targetNamespace).Get(encryptionConfSecret)
+		if errors.IsNotFound(err) {
+			recorder.Warningf("ObserveEncryptionConfigNotFound", "encryption config secret %s/%s not found", targetNamespace, encryptionConfSecret)
+			// TODO what is the best thing to do here?
+			// for now we do not unset the config as we are checking a synced version of the secret that could be deleted
+			// return observedConfig, errs
+			return previouslyObservedConfig, errs // do not append the not found error
```
[the remark this answers is not in the document]
Comment (sttts) [reply pkg/operator/encryption/observe_encryption_config.go]: put a better warning in place using  `recorder.Warningf`: `encryption config secret ... not found after encryption has been enabled`

File: pkg/operator/configobservation/configobservercontroller/observe_config_controller.go (hunk 1 above)
Comment (sttts) [reply pkg/operator/configobservation/configobservercontroller/observe_config_controller.go]: ic, name conflict

Comment (sttts) [reply pkg/operator/configobservation/configobservercontroller/observe_config_controller.go]: can this be private?

Comment (p0lyn0mial) [reply pkg/operator/configobservation/configobservercontroller/observe_config_controller.go]: field and method (`SecretLister()`) cannot be the same

File: pkg/operator/encryption/observe_encryption_config.go [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+const (
+	encryptionConfFilePath = "/etc/kubernetes/static-pod-resources/secrets/encryption-config/encryption-config"
+)
+
+type SecretLister interface {
+	SecretLister() corev1listers.SecretLister
```
[the remark this answers is not in the document]
Comment (p0lyn0mial) [reply pkg/operator/encryption/observe_encryption_config.go]: you don't like defining an interface close to a consumer?

File: pkg/operator/configobservation/interfaces.go (hunk 3 above)
Comment (p0lyn0mial) [reply pkg/operator/configobservation/interfaces.go]: okay, will take it back :)

File: pkg/operator/encryption/observe_encryption_config.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (p0lyn0mial) [reply pkg/operator/encryption/observe_encryption_config.go]: yes, so it is safe, I will remove `TODO` and add a comment.

File: pkg/operator/configobservation/configobservercontroller/observe_config_controller.go (hunk 2 above)
Comment (p0lyn0mial) [reply pkg/operator/configobservation/configobservercontroller/observe_config_controller.go]: will check :)

Comment (p0lyn0mial) [reply pkg/operator/configobservation/configobservercontroller/observe_config_controller.go]: It's the same for `openshift-apiserver` - it [takes](https://github.com/openshift/openshift-apiserver/blob/master/pkg/cmd/openshift-apiserver/openshiftapiserver/config.go#L38) [OpenShiftAPIServerConfig. APIServerArguments](https://github.com/openshift/api/blob/master/openshiftcontrolplane/v1/types.go#L45). No need to make it a param.
